README.md

```diff
@@ -157,6 +157,91 @@ bin/rails glueby:contract:timestamp:confirm
 confirmed (id=1, txid=8d602ca8ebdd50fa70b5ee6bc6351965b614d0a4843adacf9f43fedd7112fbf4)
 ```
 
+## Use fee provider mode
+
+Glueby contracts have two different way of fee provisions.
+
+1. `:sender_pays_itself`
+2. `:fee_provider_bears`
+
+The first one: `:sender_pays_itself`, is the default behavior.
+In the second Fee Provider mode, the Fee Provider module pays a fee instead of the transaction's sender.
+
+### Fee Provider Specification
+
+* Fee Provider pays fixed amount fee, and it is configurable.
+* Fee Provider needs to have enough funds into their wallet.
+* Fee Provider is managed to keep some number of UTXOs that have fixed fee value by rake tasks.
+
+### Setting up Fee Provider
+
+1. Set like below
+
+```ruby
+Glueby.fee_mode = :fee_provider_bears
+Glueby::FeeProvider.configure({
+  # The fee that Fee Provider pays on each transaction.
+  fixed_fee: 1000,
+  # Fee Provider tries to keep the number of utxo in utxo pool as this size using `glueby:fee_provider:manage_utxo_pool` rake task
+  utxo_pool_size: 20
+})
+```
```

Comment:
followin are better:

```
Glueby::FeeProvider.configure do |c|
  c.fixed_fee = 1000
  c.utxo_pool_size = 20
end
```

Reply:
How do you like to create a new configuration method for whole Glueby modules, If you want to improve the configuration ways?

like this:

```ruby
Glurby.configuration do |c|
  c.wallet_adapter = :active_record
  
  c.rpc_config = { schema: 'http', host: '127.0.0.1', port: 12381, user: 'user', password: 'pass' }
  
  # Use FeeProvider to supply inputs for fees on each transaction that is created on Glueby.
  c.fee_provider_bears!
  c.fee_provider_config = { fixed_fee: 1000, utxo_pool_size: 20 }
end
```

Reply:
LGTM

Reply:
Should I do it in this PR or create another PR?

Reply:
You can update this PR as spec. Implementation is another one.

Reply:
update spec at ddfae69